src/java/org/apache/cassandra/service/CassandraDaemon.java

```diff
@@ -426,6 +426,9 @@ public void uncaughtException(Thread t, Throwable e)
         // Native transport
         nativeTransportService = new NativeTransportService();
 
+        // load blacklisted partitions into cache
+        StorageService.instance.refreshBlacklistedPartitionsCache();
```

Comment:
I think this is where we might have to deal with some nodes not having it potentially.

Reply:
can you elaborate please? not having what?

Reply:
During a cluster upgrade, I'm not sure that we can assume this call will succeed. I'm not entirely sure how that works tbh with system_distributed tables. I guess that the first node of the new version will create the table maybe?

Reply:
yeah, I think the first node of the new version would create the table. Either way, it wouldn't be a catastrophic failure for server start since possible failure during reading blacklisted partitions is handled.